src/pages/home/home.ts

```diff
@@ -1,133 +1,233 @@
-import { Component } from '@angular/core/';
+import { Component } from "@angular/core/";
 import {
   GoogleMaps,
   GoogleMap,
   GoogleMapsEvent,
-  GoogleMapOptions,
-  CameraPosition,
-  MarkerOptions,
-  Marker,
-} from '@ionic-native/google-maps';
-import { Platform } from 'ionic-angular';
-import { YelpService } from '../../app/services/yelp/yelp.service';
-import { Observable } from 'rxjs/Observable';
-import { Geolocation } from '@ionic-native/geolocation';
-import { WorkfromService } from '../../app/services/workfrom/workfrom.service';
+  GoogleMapOptions
+} from "@ionic-native/google-maps";
+import { Platform, ModalController, NavParams } from "ionic-angular";
+import { WorkfromService } from "../../app/services/workfrom/workfrom.service";
+import { SpacePage } from "../space/space";
+import { GroupSocketService } from "../../app/services/groupsocket/groupsocket.service";
+import geolib from "geolib";
+import {UserInfo} from "../../app/services/groupsocket/groupsocket.service";
+
+interface Coordinates {
+  latitude: number;
+  longitude: number;
+}
+
+const RANDOM_GEOCOORDINATES: Coordinates[] = [
+  { latitude: 25.992046, longitude: -80.283645 }, // Pembroke Pines
+  { latitude: 25.942871, longitude: -80.12338 } // Sunny Isles
+  // { latitude: 38.5678818, longitude: -121.4636956 }, // East Sacramento
+  // { latitude: 37.2972316, longitude: -122.0976092 }, // San Jose
+];
 
 @Component({
-  selector: 'page-home',
-  templateUrl: 'home.html',
+  selector: "page-home",
+  templateUrl: "home.html"
 })
 export class HomePage {
-  yelpLocations$: Observable<any>;
-  workfromLocations$: Observable<any>;
+  map: GoogleMap;
+  mapElement: HTMLElement;
 
+  //For use outisde promise chain.
   latitude: number;
   longitude: number;
+  host_uid : string;
 
-  map: GoogleMap;
-  mapElement: HTMLElement;
+  //Random username.
+  username : string;
 
   constructor(
     private googleMaps: GoogleMaps,
     public platform: Platform,
-    private yelpService: YelpService,
     private workfromService: WorkfromService,
-    private geolocation: Geolocation
-  ) { }
+    public modalCtrl: ModalController,
+    private groupSocketService: GroupSocketService,
+    private navParams: NavParams
+  ) 
+  {
+    //Generate random username and pass to socketservice.
+    this.username = `TestUser${Math.floor(Math.random() * 100)}`;
+    this.groupSocketService.username = this.username;
+
+
+  }
+
+  ionViewWillEnter()
+  {
+    this.host_uid = this.navParams.get("group_uid")
+    this.joinHostGroup();
+
+  }
 
   ngAfterViewInit() {
-    console.log('Ion view loaded.');
+    console.log("Ion view loaded.");
+    this.groupSocketService.userInfoSubject.subscribe(userInfo => {
+      console.log(`Marker dropped for user: ${userInfo.username}`);
+      this.dropMarker(userInfo.username, "red", userInfo.latitude, userInfo.longitude );
```

Comment:
Let's not use red here for the pin of the other users. Let's use blue please. The red pins are the ones that are workfrom locations.

Reply:
Blue pin being used now